Dev

SSSenia opened the pull request.

SSSenia committed:
add: authorization by Auth0

SSSenia committed:
fix: authentication failure

SSSenia committed:
fix: removed Observables

SSSenia committed:
fix: changed function calls to pipes

SSSenia committed:
fix: short form notation

SSSenia committed:
fix: constants changed

SSSenia committed:
fix: BEM, access modifiers
*html and css replaced according to BEM methodology
*added access modifiers for methods and properties

santa-nyan commented on src/app/meal/edit-meal-page/edit-meal-page.component.scss:
```diff
@@ -5,70 +5,71 @@
     max-width: 98vmin;
```
В этом файле и во всех остальных есть дублированние стилей, например, здесь есть такие же как в new-meal-page.component.scss, и в таком случае ты должен вынести их либо в глобальные стили, либо создать отдельный файл в котором будут эти стили, а потом просто добавлять его в декоратор нужных компонентов в свойство styleUrls

santa-nyan commented on src/app/meal/edit-meal-page/edit-meal-page.component.scss:
```diff
@@ -5,70 +5,71 @@
+            width: 100%;
 
-            .time {
+            ::ng-deep .mat-input-element {
```
здесь ты используешь ng-deep, который уже deprecated достаточно давно, вот скрин с оф доки Ангуляра
![image](https://user-images.githubusercontent.com/44004086/195329501-fee4cc77-d199-4a09-9c1c-ba31a92a7aa2.png)
так что - на этот счет существует два решения, либо опять же, вынести такие штуки на глобальный уровень и просто использовать имя материал класса (без ng-deep), либо окунуться в custom material theming, что я настоятельно рекомендую сделать, ибо это наилучший вариант, и отредачить таким образом ВСЕ scss файлы где такое используется, не только данный файл, где я оставил коммент

santa-nyan commented on src/app/app-routing.module.ts:
```diff
@@ -4,14 +4,16 @@ import { StartPageComponent } from './start-page/start-page.component';
+  { path: 'calendar', component: CalendarPageComponent, canActivate: [AuthGuard] },
+  { path: 'setup', component: SetupPageComponent, canActivate: [AuthGuard] },
+  { path: 'view-day', component: ViewDayPageComponent, canActivate: [AuthGuard] },
+  { path: 'meal', loadChildren: () => import('./meal/meal.module').then(m => m.MealModule), canActivate: [AuthGuard] },
```
здесь можно попробовать реализовать все по новому стандарту, ибо в новой версии Ангуляра подьехала фича canMatch, и ее можно было бы использовать здесь, вот подробный видос 
https://www.youtube.com/watch?v=OpBFhnLlhdE

santa-nyan commented on src/app/shared/guards/auth.guard.ts:
```diff
@@ -0,0 +1,19 @@
+    public canActivate(): boolean | Observable<boolean> | Promise<boolean> {
+        return this.auth.isAuthenticated$.pipe(
+            tap((value) => {
+                if (!value) this.router.navigate(['/start'])
```
здесь через Router можно сделать красивее редирект - используя router.createUrlTree() вместо router.navigate, ибо первый создан как раз для таких случаев

я уже скидывал этот ролик в другом комменте, но тут тоже явно показано, как можно использовать createUrlTree
https://www.youtube.com/watch?v=OpBFhnLlhdE

santa-nyan reviewed (commented):
глянуть на оставленные комменты

SSSenia committed:
fix: Two identical scss files merged into one

SSSenia committed:
fix: changed guards to urlTree and setted canMatch

SSSenia committed:
fix: Replaced ::ng-deep with global styles
Replaced ::ng-deep with global styles for the angular material in:
*calendar-page.scss
*input-meal.scss
